group/hash.go

```diff
@@ -0,0 +1,89 @@
+package group
+
+import (
+	"crypto"
+	"math/big"
+)
+
+// GFHasher allows to hash byte strings producing one or more field elements.
+type GFHasher interface {
```

Comment:
Does this need to be exported? Also, perhaps this should be `FieldHasher`, rather than `GFHasher`?